src/commands/privmsg.cpp

```diff
@@ -6,15 +6,241 @@
 /*   By: faboussa <faboussa@student.42lyon.fr>      +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/11/04 10:18:52 by yusengok          #+#    #+#             */
-/*   Updated: 2024/11/12 11:53:58 by faboussa         ###   ########.fr       */
+/*   Updated: 2024/11/15 18:29:51 by faboussa         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
 #include "../../includes/Server.hpp"
 
+// PRIVMSG <target>{,<target>} <:text to be sent>
+// <target> is the nickname of a client or the name of a channel.
+
+void Server::sendPrivmsgToClient(const Client &sender, const Client &receiver,
+                                 const std::string &content) {
+  std::string message = ":" + sender.getNickname() + " PRIVMSG " +
+                        receiver.getNickname() + " " + content + "\r\n";
+  receiver.receiveMessage(message);
+}
+
+// Fonction pour valider les arguments de privmsg
+bool Server::validTargets(const std::string &arg, const Client &client,
+                          stringVector *targets) {
+  size_t commaCount = std::count(arg.begin(), arg.end(), ',');
+#ifdef DEBUG
+  std::cout << "commaCount: " << commaCount << std::endl;
+  std::cout << "MAXTARGETS: " << gConfig->getLimit(MAXTARGETS) << std::endl;
+#endif
+  if (commaCount + 1 > gConfig->getLimit(MAXTARGETS)) {
+    send407TooManyTargets(client);
+    return false;
+  }
+  // verifier que chaque cible est homogene. pas de mix @ et # et user
+
+  if (commaCount != 0) {
+    size_t chanOpPrefixCount = 0;
+    size_t channelsCount = 0;
+    size_t clientsCount = 0;
+
+    stringVector::iterator itEnd = (*targets).end();
+    for (stringVector::iterator it = (*targets).begin(); it != itEnd; ++it) {
+      std::string target = *it;
+#ifdef DEBUG
+      std::cout << "target: " << target << std::endl;
+#endif
+      if (target[0] == CHAN_OP)
+        chanOpPrefixCount++;
+      else if (target[0] == REG_CHAN)
+        channelsCount++;
+      else
+        clientsCount++;
+    }
+    if (commaCount != 0 && channelsCount != commaCount + 1 &&
+        clientsCount != commaCount + 1 && chanOpPrefixCount != commaCount + 1) {
+      send407TooManyTargets(client);
+      return false;
+    }
+  }
+  return true;
+}
+
+bool Server::parseArguments(const std::string &arg, const Client &client,
+                            std::vector<std::string> *targetsVector,
+                            std::string *message) {
```

Comment:
**[CHANEGED]** Rename function to `parsePrivmsgArguments` to avoid confusion because this helper function is specific for PRIVMSG but attached to Server class.